migration/v0.42/apply/10-upgrade-grafana.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+ROOT="$(readlink -f "$(dirname "${0}")/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+run() {
+  case "${1:-}" in
+  execute)
+    if [[ "${CK8S_CLUSTER}" =~ ^(sc|both)$ ]]; then
+      log_info "Deleting existing Grafana releases to address volumeName templating issue"
+
+      mapfile -t helm_secrets < <(kubectl_do sc get secrets -n monitoring -oname | grep -E 'sh.helm.release.v1.ops|sh.helm.release.v1.user-grafana')
+
+      if [[ "${#helm_secrets[@]}" -gt 0 ]]; then
+        log_info "The following secrets will be deleted:"
+        printf '%s\n' "${helm_secrets[@]}"
+        log_info_no_newline "Do you want to continue? (Y/n): "
+        read -r reply
+        if [[ ! "${reply:-y}" =~ ^[yY]$ ]]; then
+          exit 1
+        fi
```

Comment:
This will make it fail in automation.
Instead, please check if the chart version in the cluster is affected, and only then perform the deletion.